Dockerfile

```diff
@@ -40,11 +41,23 @@ RUN sudo mkdir -p /opt/opam/doc && sudo chown -R opam:opam /opt/opam
 RUN cp -r doc/html /opt/opam/doc/api
 RUN cp -r doc/man-html /opt/opam/doc/man
 RUN cp -r doc/pages/* /opt/opam/doc/
+
+FROM --platform=linux/amd64 ocaml/opam:archive as opam-archive
+FROM ocaml/opam.ocaml.org-legacy as opam-legacy
 FROM alpine:3.14 as opam2web
-RUN apk add git curl rsync libstdc++
+RUN apk add --update git curl rsync libstdc++ rdfind
+COPY --from=opam-legacy . /www
 COPY --from=build-opam2web /opt/opam2web /usr/local
 COPY --from=build-opam-doc /usr/bin/opam /usr/local/bin/opam
 COPY --from=build-opam-doc /opt/opam/doc /usr/local/share/opam2web/content/doc
+RUN --mount=type=bind,target=/cache,from=opam-archive rsync -aH /cache/cache/ /www/cache/
+COPY ext/key/opam-dev-team.pgp /www/opam-dev-pubkey.pgp
```

Comment:
Is that key kept on the deployer?

Reply:
The public opam key(opam-dev-team.pgp) is copied from this git repo into the container, on the deployer. So it will exist here, on the deployer as a git checkout and on the docker image.

That should be ok since it's a public key.